tracing-distributed/src/telemetry_layer.rs

```diff
@@ -237,7 +237,7 @@ where
                         parent_id: Some(self.trace_ctx_registry.promote_span_id(parent_id)),
                         initialized_at,
                         meta: event.metadata(),
-                        service_name: &self.service_name,
+                        service_name: self.service_name,
```

Comment:
Note to self: is this change needed? See also https://github.com/eaze/tracing-honeycomb/pull/13#discussion_r772765094

Reply:
This change *is* required - rust will build fine without this change but clippy gets irate.

I was low key concerned that this change would have backwards incompatible ramifications, *but* we'll be rolling out a breaking semver anyway so it's more or less moot.